js/src/contract/item.ts

```diff
@@ -421,869 +42,283 @@ export class ItemProgram {
-      args.classIndex
+    args: ItemInstruction.StartItemEscrowBuildPhaseArgs,
+    accounts: ItemInstruction.StartItemEscrowBuildPhaseAccounts,
+    additionalArgs: ItemInstruction.StartItemEscrowBuildPhaseAdditionalArgs,
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const instruction = await this.instruction.startItemEscrowBuildPhase(
+      args,
+      accounts,
+      additionalArgs
     );
-
-    await this.program.methods
-      .createItemClass(args)
-      .accounts({
-        itemClass: itemClassKey,
-        itemMint: accounts.itemMint,
-        metadata: await getMetadata(accounts.itemMint),
-        edition: await getEdition(accounts.itemMint),
-        parent: accounts.parent || itemClassKey,
-        payer: (this.program.provider as AnchorProvider).wallet.publicKey,
-        systemProgram: SystemProgram.programId,
-        rent: web3.SYSVAR_RENT_PUBKEY,
-      })
-      .remainingAccounts(remainingAccounts)
-      .rpc();
-
-    return itemClassKey;
+    return this.sendWithRetry(instruction, [], options);
   }
 
   async updateItem(
-    args: UpdateItemArgs,
-    _accounts: UpdateItemAccounts,
-    additionalArgs: UpdateItemAdditionalArgs
-  ): Promise<web3.PublicKey> {
-    const itemClassKey = (
-      await getItemPDA(args.itemClassMint, args.classIndex)
-    )[0];
-
-    const itemKey = (await getItemPDA(args.itemMint, args.index))[0];
-
-    await this.program.methods
-      .updateItem(args)
-      .accounts({
-        accounts: {
-          itemClass: itemClassKey,
-          item: itemKey,
-        },
-      })
-      .rpc();
-
-    return itemClassKey;
+    args: ItemInstruction.UpdateItemArgs,
+    accounts: ItemInstruction.UpdateItemAccounts,
+    additionalArgs: ItemInstruction.UpdateItemAdditionalArgs,
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const instruction = await this.instruction.updateItem(
+      args,
+      accounts,
+      additionalArgs
+    );
+    return this.sendWithRetry(instruction, [], options);
   }
+}
 
-  async updateItemClass(
-    args: UpdateItemClassArgs,
-    accounts: UpdateItemClassAccounts,
-    additionalArgs: UpdateItemClassAdditionalArgs
-  ): Promise<web3.PublicKey> {
-    const remainingAccounts = additionalArgs.permissionless
-      ? [{ pubkey: accounts.parent, isWritable: false, isSigner: false }]
-      : await generateRemainingAccountsGivenPermissivenessToUse({
-          permissivenessToUse: args.updatePermissivenessToUse,
-          tokenMint: accounts.itemMint,
-          parentMint: accounts.parentMint,
-          parentIndex: args.parentClassIndex,
-          parent: accounts.parent,
-          metadataUpdateAuthority: accounts.metadataUpdateAuthority,
-          program: this.program,
-        });
-
-    convertNumsToBNs(args);
-
-    const itemClassKey = (
-      await getItemPDA(accounts.itemMint, args.classIndex)
-    )[0];
+interface SendOptions {
```

Comment:
We should move this into sol-kit.

Reply:
Probably worth it to also have another type for the responses too:
```
{
     txid: string;
     slot: number;
 }
```

Reply:
Added here: https://github.com/raindrop-studios/sol-kit/pull/2